File: metnum.py
import math
import time
start = time.time() # untuk menghitung waktu

print("-"*25 + "NEWTON RAPHSON METHOD" + "-"*25) # kode untuk metode newton raphson dimulai

x0 = 10
raem = 100.0
i = 0

while raem>=1 :
    print("-" * 35)
    print("Iterasi i=",i)
    i = i + 1
    print("x0      :", x0)
    f = (1.5 ** x0) - 5
    fd1 = (1.5 ** x0) * 0.4054651081
    x1 = (x0) - (f/fd1)
    print("x1      :", x1)
    rae = ((x1-x0)/x1) * 100
    if rae<0:
        raem = rae * (-1)
    else:
        raem = rae
    print("rae     :", raem)
    x0 = x1

print(" ")
print(" ")

print("-"*25 + "SECANT METHOD" + "-"*25) # kode untuk metode secant dimulai

X0 = 10
X1 = 8
raem = 100.0
i = 0

while raem>=1:
    print("-" * 35)
    print("Iterasi i=",i)
    i = i + 1
    print("x0      :", X0)
    print("x1      :", X1)
    fx0 = (1.5 ** X0) - 5
    fx1 = (1.5 ** X1) - 5
    X2 = X1 - ((fx1)*(X1-X0)/(fx1-fx0))
    print("x2      :", X2)
    rae = ((X2-X1)/X2)*100
    if rae<0:
        raem = rae * (-1)
    else:
        raem = rae
    print("rae     :", raem)
    X0 = X1
    X1 = X2

end = time.time()
print(" ")
print("Runtime :", end - start)

File: metnum.c
//program metode numeris newton raphson method dan secant method
#include <stdio.h>
#include <math.h>
#include <time.h>

void newtonraphson();
void secant();

int main(){

    clock_t start, end; // untuk menghitung runtime
    double cpu_time_used;
    start = clock();
    printf("\n");
    newtonraphson();
    printf("\n");
    secant();
    printf("\n");
    end = clock();
    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
    printf("Runtime  : %lf", cpu_time_used);

    return 0;

}

void newtonraphson(){

    float x=10, fx;
    float fd1, xa, rae, raem=100;

    printf("-------------------------NEWTON RAPHSON METHOD-------------------------");
    printf("\n");

    for(int i=0; raem>=1; i++){ // coding metode newton raphson start
    
        printf("Iterasi i=%d ------------------------------------------\n", i);
        printf("x0       : %f \n", x);
        fx = pow((1.5),x)-5; //rumus utama
        fd1 = pow((1.5),x) * 0.4054651081; // <<< mulai algoritma newton rhapson
        xa = x-(fx/fd1);
        printf("x1       : %f \n", xa);
        rae = ((xa-x)/xa)*100;
        if(rae<0){
            raem = rae*(-1);
        }
        else{
            raem = rae;
        }
        printf("rae      : %f \n", raem);
        x = xa;

    }

    printf("\n");

}

void secant(){

    float x0=10, x1=8, fx0, fx1;
    float x2, rae, raem=100;

    printf("-------------------------SECANT METHOD-------------------------");
    printf("\n");

    for(int i=0; raem>=1; i++){ // coding metode secant start

        printf("Iterasi i=%d ------------------------------------------\n", i);
        printf("x0       : %f \n", x0);
        printf("x1       : %f \n", x1);
        fx0 = pow((1.5),x0)-5; //rumus utama dimasukin x0
        fx1 = pow((1.5),x1)-5; //rumus utama dimasukin x1
        x2 = x1 - ((fx1 * (x1-x0))/(fx1 - fx0)); // <<< mulai algoritma secant
        printf("x2       : %f \n", x2);
        rae = ((x2-x1)/x2)*100;
        if(rae<0){
            raem = rae*(-1);
        }
        else{
            raem = rae;
        }
        printf("rae      : %f \n", raem);\
        x0 = x1;
        x1 = x2;

    }

}